app/src/main/java/com/example/concordia_campus_guide/Models/Direction.java

```diff
@@ -2,21 +2,28 @@
 
 import com.example.concordia_campus_guide.Models.Routes.TransportType;
 import com.google.android.gms.maps.model.LatLng;
+
+import java.io.Serializable;
 import java.util.Date;
 
-public class Direction {
-    private LatLng start;
-    private LatLng end;
+public class Direction implements Serializable {
+    private transient LatLng start;
+    private transient LatLng end;
     private Date startTime;
     private Date endTime;
-    private TransportType transportType;
+    private String transportType;
     private String description;
+    private double duration;
+
+    public Direction() {
+    }
 
-    public Direction(LatLng start, LatLng end, TransportType transportType, String description) {
+    public Direction(LatLng start, LatLng end, String transportType, String description, double duration) {
         this.start = start;
         this.end = end;
         this.transportType = transportType;
```

Comment:
We have a class called transportType already n now we have this transportType string here which is leading to confusion here which represents what exactly.

Reply:
yes that's my bad, I should have a different name for the class TransportType and for the constant TransportType.

Reply:
If this is the StringDef then we could add the annotation `@ClassConstants.TransportType` in front of the String.

Reply:
I will resolve this conflict for now... we can add an issue for it